permission_handler_html/lib/web_delegate.dart

```diff
@@ -77,8 +74,8 @@ class WebDelegate {
       // The manual stop action is then needed here for to stop the automatic
       // recording.
 
-      if (mediaStream.active!) {
-        final audioTracks = mediaStream.getAudioTracks();
+      if (mediaStream?.active ?? false) {
+        final audioTracks = mediaStream!.getAudioTracks();
         if (audioTracks.isNotEmpty) {
           audioTracks[0].stop();
         }
```

Comment:
Combining the condition here mixes two different situations and treating them the same, which leads to unexpected behaviour.

1. The original test is just to make sure the `mediaStream` is stopped if it is started (which is caused by calling the `getUserMedia()` method on line 68).
2. The added tests checks if `mediaStream` is `null`, which is most likely the result of `_devices` being `null`. 

I feel we should `return false;` at the start of the method if `_devices` is `null` and prevent this mixed behaviour. Added benefit will be that returning `false` is translated to a `PermissionStatus.denied` which is was it comes down to in a HTTP (non-secure) context.

Reply:
> 2\. The added tests checks if `mediaStream` is `null`, which is most likely the result of `_devices` being `null`

Not only. The `mediaStream.active` value is also nullable. Though a good catch on `PermissionStatus.denied`. 

The original behaviour to return `true` by default is now how I'd do it, but I didn't want to completely rewrite this. Perhaps someone could improve it when doing refactoring to remove dependency on `html` packages in order to support wasm...